hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java

```diff
@@ -1788,7 +1787,13 @@ void handleMapNode(Node parentNode, final MapConfig mapConfig) throws Exception
             } else if ("async-backup-count".equals(nodeName)) {
                 mapConfig.setAsyncBackupCount(getIntegerValue("async-backup-count", value));
             } else if ("eviction".equals(nodeName)) {
-                mapConfig.setEvictionConfig(getEvictionConfig(node, false, true));
+                EvictionConfig evictionConfig = getEvictionConfig(node, false, true);
```

Comment:
I was actually thinking if it would not make sense that we validate the whole map config, just like we do when creating a new distributed object, like this:
```
    void handleMapNode(Node parentNode, final MapConfig mapConfig) throws Exception {
        for (Node node : childElements(parentNode)) {
            ...
        }
        try {
            checkMapConfig(mapConfig, config.getNativeMemoryConfig(), null, new HazelcastProperties(config));
        } catch (IllegalArgumentException e) {
            throw new InvalidConfigurationException(e.getMessage());
        }
        config.addMapConfig(mapConfig);
    }
```
Not sure if there's anything preventing us from doing so cc @ahmetmircik .

Reply:
@mmedenjak do you mean put down `try` `catch` block at the end of the `handleMapNode` method?

Reply:
Yes, I was thinking if we can (and if we can't, why) have just one method exposed on `ConfigValidator` that validates the entire map config and use it whenever the whole map config is either parsed or when we go to create a new distributed object.

Reply:
I've updated the code according to the discussion